### PR TITLE
Initial Framework Design

### PR DESCRIPTION
**Pull request by raveerocks**
The commit contains the following

1. The JUnit5 WebDriverTest extension
2. Configuration files
3. Profiles
4. Tests
5. Resources

### CODE REVIEW

**Comment by sanketsmali03** [comment]
@raveerocks  : Please create a [jenkins job ](https://customer-demos.bsstag.com/) as it is one of the steps to be implemented for Demo repository

File: src/test/java/com/browserstack/examples/config/WebDriverConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.List;
+
+public class WebDriverConfiguration {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(WebDriverConfiguration.class);
```
Comment (sanketsmali03) [inline src/test/java/com/browserstack/examples/config/WebDriverConfiguration.java:12]: The Logger object has not been used in the class

Comment (raveerocks) [reply src/test/java/com/browserstack/examples/config/WebDriverConfiguration.java:12]: @sanketsmali03 https://github.com/browserstack/browserstack-examples-junit5/pull/7#issuecomment-889748455 is resolved.

**Comment by sanketsmali03** [comment]
@raveerocks  : 

- Add provision to run profile which executes a specific test case for maven & gradle
- Confirm readme template 
- User is required to install allure command line tool seperately to generate allure report , can we handle this directly via gradle or maven allure dependency
- Mark on testcase as fail > Login as User with no image loaded : This is to have an example of failed test in the repo
- Test case name are not as per the expected format

**Pull request closed without merging**